interface/controller/pkmn_controller.go

```diff
@@ -0,0 +1,49 @@
+package controller
+
+import (
+	"net/http"
+
+	"github.com/jvillarreal-w/academy-go-q42021/domain/model"
+	"github.com/jvillarreal-w/academy-go-q42021/usecase/interactor"
+)
+
+type pokemonController struct {
+	pokemonInteractor interactor.PokemonInteractor
+}
+
+type PokemonController interface {
+	GetPokemon(c Context) error
+	GetPokemonById(c Context) error
+}
+
+func NewPokemonController(pi interactor.PokemonInteractor) PokemonController {
+	return &pokemonController{pi}
+}
+
+func (pc *pokemonController) GetPokemon(c Context) error {
+	var p []*model.Pokemon
+
+	p, err := pc.pokemonInteractor.Get(p)
+	if err != nil {
+		return err
+	}
+
+	return c.JSON(http.StatusOK, p)
+}
+
+func (pc *pokemonController) GetPokemonById(c Context) error {
+	var p *model.Pokemon
+	id := c.Param("id")
```

Comment:
One of the main responsibilites of a controller is to validate if the data you receive from the user is valid before moving on, if you need to convert this to an int later on, the place to do it is here on your controller, and not in deeper layers of your app. So if the id is not valid, you could return an error response to the user indicating that the id provided is not valid

Reply:
Now validated in the controller.